scripts/metadata_comparison/metadata_comparison/extractor.py

```diff
@@ -22,40 +23,62 @@
+    workflow_regex_validator
 from metadata_comparison.lib.operation_ids import get_operation_id_number, find_operation_ids_in_metadata
 from metadata_comparison.lib.papi.papi_clients import PapiClients
-from typing import Mapping, Any
+from metadata_comparison.lib.storage import upload_blob
+from metadata_comparison.lib.logging import set_log_verbosity, quieten_chatty_imports
+from typing import Mapping, Any, Union
 
 logger = logging.getLogger('metadata_comparison.extractor')
 
 
-def set_log_verbosity(verbose: bool) -> None:
-    if verbose:
-        logging.basicConfig(format='[%(asctime)s] [%(name)s] %(message)s', level=logging.INFO)
-    else:
-        logging.basicConfig(format='[%(asctime)s] [%(name)s] %(message)s', level=logging.WARNING)
+def __create_snapshot_of_local_repo(repo: git.Repo, cromwell_snapshots_path: Union[Path, Any]) -> Union[Path, Any]:
+    last_commit_hash = repo.head.commit.hexsha
+    if not os.path.exists(cromwell_snapshots_path):
+        os.makedirs(cromwell_snapshots_path)
+    current_snapshot_path = cromwell_snapshots_path / last_commit_hash
+    if not os.path.exists(current_snapshot_path):
+        os.makedirs(current_snapshot_path)
+        repo.clone(current_snapshot_path)
+    return current_snapshot_path
 
 
-def quieten_chatty_imports() -> None:
-    logging.getLogger('googleapiclient.discovery_cache').setLevel(logging.ERROR)
-    logging.getLogger('googleapiclient.discovery').setLevel(logging.WARNING)
+def __create_zip_file(zip_file_path: Union[Path, Any], current_snapshot_path: Union[Path, Any]):
+    with zipfile.ZipFile(zip_file_path, "a", allowZip64=False) as zip_file:
+        for root, dirs, files in os.walk(current_snapshot_path):
+            for file in files:
+                zip_file.write(os.path.join(root, file))
 
 
-def upload_local_checkout() -> None:
-    #   # Make a snapshot of the local Cromwell git repo
-    #
-    #   hash=$(git rev-parse HEAD)
-    #
-    #   # Make sure all files are checked in.
-    #   if [[ $(git status --short | wc -l) -ne 0 ]]; then echo "stuff needs to be checked in"; exit 1; fi
-    #
-    #   git_snapshots_dir="../git-snapshots"
-    #   mkdir -p "${git_snapshots_dir}"
-    #   git_snapshot="${git_snapshots_dir}/$hash"
-    #   git clone . --local "${git_snapshot}"
-    raise Exception("Not Implemented")
+def upload_local_checkout(cromwell_path: Path,
+                          gcs_bucket: str,
+                          gcs_path: str,
+                          gcs_storage_client: storage.Client) -> None:
+    cromwell_snapshots_path = cromwell_path.parent / "cromwell_snapshots"
+
+    repo = git.Repo(cromwell_path)
+    if repo.is_dirty():
+        raise Exception("Unable to upload local checkout to GCS: repository is dirty - need to do check in first.")
+
+    last_commit_hash = repo.head.commit.hexsha
+    zip_file_name = f"{last_commit_hash}.zip"
+    zip_file_path = Path(cromwell_snapshots_path / zip_file_name)
```

Comment:
I believe the snapshots path would be under the workflow id so I think there should be only one snapshot per workflow. If so a standardized name for the snapshot might be helpful for any downstream code that wants to consume it.